pkg/tikv/client.go

```diff
@@ -29,21 +31,32 @@ type Client struct {
 	httpScheme   string
 	lifecycleCtx context.Context
 	timeout      time.Duration
+	isRawBody    bool
+	getEndpoints func() (map[string]struct{}, error)
 }
 
-func NewTiKVClient(lc fx.Lifecycle, httpClient *httpc.Client, config *config.Config) *Client {
+func NewTiKVClient(lc fx.Lifecycle, httpClient *httpc.Client, pdClient *pd.Client, config *config.Config) *Client {
+	cache := pd.NewEndpointCache()
 	client := &Client{
 		httpClient:   httpClient,
 		httpScheme:   config.GetClusterHTTPScheme(),
 		lifecycleCtx: nil,
 		timeout:      defaultTiKVStatusAPITimeout,
+		getEndpoints: func() (map[string]struct{}, error) {
+			return cache.Func("tikv_endpoints", func() (map[string]struct{}, error) {
```

Comment:
I guess you will find your code logic even more simpler without using the TTLCache, as:

1. You will only store one item
2. You want to have strong type checks